src/drivers/localstorage.js

```diff
@@ -123,7 +124,7 @@
                     value = serializer.deserialize(value);
                 }
 
-                value = iterator(value, key.substring(keyPrefixLength), i + 1);
+                value = iterator(value, key.substring(keyPrefixLength), iterationNumber++);
```

Comment:
I think that the test do not suffice only because of the internal implementation of locastorage (probably browser specific).
It seems that even though `localStorage.setItem('local', 'forage');` is executed first, `iterate` returns it last !?
In my tests in chrome (on this exact page if that matters) `localStorage.key(0)` returns the last item inserted, like a stack.
Adding an  extra item in localStorage right after `localforage.setItem('name', 'Bob');` resolves should be enough.